File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp
#include<iostream>
#include<malloc.h>
using namespace std;
struct node
{
    int data;
    struct node *next;
};
void Traverse_stack(struct node* top)
{
    struct node* temp=top;
    cout<<"Current Stack->\n";
    while(temp!=NULL)
    {
    cout<<temp->data<<"\n";
    temp=temp->next;
    }
}
struct node * push(struct node * top,int data)
{
    struct node * temp;
    temp=(struct node*)malloc(sizeof(struct node));
    temp->data=data;
    temp->next=top;
    top=temp;
    cout<<top->data<<" is pushed into the stack\n";
    Traverse_stack(top);
    return top;
}
struct node* pop(struct node* top)
{
    struct node * temp;
    temp=top;
    if(top==NULL)
    {
            cout<<"Stack is empty\n";

    }
    else
    {
      top=top->next;
      cout<<temp->data<<"is popped out of the stack\n";
      free(temp);
      Traverse_stack(top);
      return top;
  }
}
void TOP(struct node * top)
{
    if(top==NULL)
    {
        cout<<"Stack is empty\n";
    }
    else
    {
        cout<<"Top->"<<top->data<<"\n";
    }

}
bool isEmpty(struct node*top)
{

  if(top==NULL)
  {
      cout<<"Stack is empty\n";
      return true;
  }
  else
  {
      return false;
  }
}


int main()
{
    struct node * top=NULL;
    top=push(top,1);
    top=push(top,2);
    top=push(top,3);

    TOP(top);

    top=pop(top);
    top=pop(top);
    top=pop(top);
    isEmpty(top);


}

File: code/data_structures/src/stack/stack/stack_using_linked_list.py
class node:
    def __init__(self,value):
        self.data=value
        self.next=None
class stack:
    def __init__(self,value):
        self.ptr=None
    def display(self):
        temp=self.ptr
        while temp:
            print (temp.data,end='->')
            temp=temp.next
        print('NULL')
    def insertnode(self,data):
        if self.ptr==None:
            self.ptr=node(data)
        else:
            newnode=node(data)
            newnode.next=self.ptr
            self.ptr=newnode
    def deletenode(self):
        if self.ptr==None:
            print("Stack is empty")
            return
        else:
            print("Item deleted is %s"%self.ptr.data)
            self.ptr=self.ptr.next
    def isEmpty(self):
        return self.ptr==None

top=stack()
while True:
    choice=int(input("1.Push\n2.Pop\n3.Isempty\n4.Exit\n"))
    if choice==1:
        element=int(input("Enter the element to be inserted: "))
        top.insertnode(element)
        top.display()
    if choice==2:
        top.deletenode()
        top.display()
    if choice==3:
        print(top.isEmpty())
    if choice==4:
        break
        
